### PR TITLE
Instagram v2

### PR DESCRIPTION
**Pull request by whiteshadow-gr**
Migrated to the new instagram api

### CODE REVIEW

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/authProviders/InstagramProvider.scala [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+            throw new ProfileRetrievalException(SpecifiedProfileError.format(id, code, errorType, errorMsg))
+          case _ => profileParser.parse(json, refreshedTokenAuthInfo)
+        }
+      }
+    }
+  }
+
+  private def getLongLivedToken(authInfo: OAuth2Info): Future[OAuth2Info] = {
+    val clientSecret = sys.env("API_CLIENT_SECRET")
```
Comment (augustinas) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/authProviders/InstagramProvider.scala]: Direct access to the environment variables is discouraged. Use configuration file to pass environment variables.

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/authProviders/InstagramProvider.scala [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import com.mohiva.play.silhouette.impl.providers._
+import InstagramProvider._
+import play.api.libs.json._
+
+import scala.concurrent.Future
+
+/**
+ * Base Instagram OAuth2 Provider.
+ *
+ * @see https://monzo.com/docs/#authentication
```
Comment (augustinas) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/authProviders/InstagramProvider.scala]: ???

File: dataplug-instagram/conf/evolutions/dataplug.sql [hunk 3]
```diff
@@ -7,3 +7,9 @@ VALUES
   ('feed', 'User feed''s media', 'sequence'),
   ('profile', 'User''s Instagram profile information', 'snapshots')
   ON CONFLICT (name) DO NOTHING;
+
+--changeset dataplugInstagram:deleteEndpoints context:data
```
Comment (augustinas) [inline dataplug-instagram/conf/evolutions/dataplug.sql]: Can we archive all of these endpoints into a separate table before deleting?

File: dataplug-instagram/conf/evolutions/dataplug.sql [hunk 4]
```diff
@@ -7,3 +7,9 @@ VALUES
...
+
```
Comment (augustinas) [inline dataplug-instagram/conf/evolutions/dataplug.sql]: Assuming Instagram authentication tokens and/or user IDs have changed as well, we should clean up the authentication token table as well.

File: dataplug-instagram/conf/logback.xml [hunk 5]
```diff
@@ -30,16 +30,22 @@
   <logger name="com.hubofallthings.dataplugInstagram.apiInterfaces.*" level="INFO" />
   <logger name="com.hubofallthings.dataplugInstagram.controllers.Api" level="INFO" />
   <logger name="com.hubofallthings.dataplugInstagram.apiInterfaces.InstagramFeedInterface" level="DEBUG" />
+  <logger name="com.hubofallthings.dataplugInstagram.apiInterfaces.InstagramProfileCheck" level="DEBUG" />
```
Comment (augustinas) [inline dataplug-instagram/conf/logback.xml]: Not comfortable with deploying a sandbox version with DEBUG mode on across the board. Great for debugging locally, but deployments (and by extension dev branch version) have to have minimum INFO level on all classes.

File: dataplug-instagram/conf/silhouette.conf [hunk 6]
```diff
@@ -5,7 +5,7 @@ silhouette {
   authenticator.cookiePath="/"
   authenticator.secureCookie=false // Disabled for testing on localhost without SSL, otherwise cookie couldn't be set
   authenticator.httpOnlyCookie=true
-  authenticator.useFingerprinting=true
```
Comment (augustinas) [inline dataplug-instagram/conf/silhouette.conf]: curious why was this change necessary?

Comment (whiteshadow-gr) [reply dataplug-instagram/conf/silhouette.conf]: I was just testing to see if these settings make a difference for the bad signature exception. Answer is nope

File: dataplug-instagram/conf/evolutions/dataplug.sql (hunk 3 above)
Comment (augustinas) [reply dataplug-instagram/conf/evolutions/dataplug.sql]: Let's follow a convention of naming archive tables with `_archive` suffix. Please change `dataplug_user_old` -> `dataplug_user_archive`.
